pkg/cluster/internal/providers/docker/network.go

```diff
@@ -48,27 +48,54 @@ func ensureNetwork(name string) error {
 	if err != nil {
 		return err
 	}
+	// network already exists, we're good
 	if exists {
 		return nil
 	}
 
-	// generate unique subnet per network based on the name
+	// Generate unique subnet per network based on the name
 	// obtained from the ULA fc00::/8 range
-	// make N attempts with "probing" in case we happen to collide
-	for i := int32(0); i < 3; i++ {
-		subnet := generateULASubnetFromName(name, i)
+	// Make N attempts with "probing" in case we happen to collide
+	subnet := generateULASubnetFromName(name, 0)
+	err = createNetwork(name, subnet)
+	if err == nil {
+		// Success!
+		return nil
+	}
+
+	// On the first try check if ipv6 fails entirely on this machine
+	// https://github.com/kubernetes-sigs/kind/issues/1544
+	// Otherwise if it's not a pool overlap error, fail
+	// If it is, make more attempts below
+	if isIPv6UnavailableError(err) {
+		// only one attempt, IPAM is automatic in ipv4 only
+		return createNetwork(name, "")
+	} else if isPoolOverlapError(err) {
```

Comment:
If ! We return, is missing the ! , No?

Reply:
Yeah, this was missed in review.
We should fix this, but it just means the probing for alternative space doesn't work which while unfortunate just means users should precreate the network with a different subnet if the one we pick is taken by them for something else.